core/frontend/src/extension/Extension.ts

```diff
@@ -32,47 +37,101 @@ export interface ExtensionManifest {
+  readonly activationEvents: ActivationEvent[];
 }
 
-/** Describes an Extension that has already been downloaded and has a location files can be easily executed.
+/**
+ * A "ready to use" Extension (contains a manifest object and a function to execute).
+ * Will be used as the type for in-memory extensions in the ExtensionAdmin
  * @alpha
-*/
-export interface LocalExtensionProps {
-  readonly manifest: ExtensionManifest;
-  readonly mainFunc?: ResolveFunc;
+ */
+export interface InstalledExtension {
+  /** A function that executes the main entry point of the extension */
+  execute: ResolveFunc;
+  /** The manifest (package.json) of the extension */
+  manifest: ExtensionManifest;
 }
 
-/** The minimum information required to download an Extension
+/**
+ * An Extension Provider defines how to fetch and execute an extension.
+ * An extension can be one of three kinds:
+ *   1. A locally installed extension.
+ *   2. A remote extension served from a user provided host.
+ *   3. A remote extension served from Bentley's Extension Service.
+ * All three must have these required properties and methods.
  * @alpha
-*/
-export interface ExtensionLoaderProps {
-  name: string;
-  version: string;
+ */
+export interface ExtensionProvider {
+  /** A function that returns the extension's manifest file */
+  getManifest: ResolveManifestFunc;
+  /** A function that executes the main entry point of the extension */
+  execute: ResolveFunc;
+  /** Hostname of a remote extension */
+  readonly hostname?: string;
+}
+
+/**
+ * Required props for a local extension provider
+ * @alpha
+ */
+export interface LocalExtensionProviderProps {
+  /** A promise that returns the manifest (package.json) of a local extension */
+  manifestPromise: Promise<any>;
+  /** A function that runs the main entry point of the local extension */
+  main: ResolveFunc;
 }
 
-/** @alpha */
-export interface BuiltInExtensionLoaderProps {
-  manifest: Promise<any>;
-  loader: ResolveFunc;
+/**
+ * Required props for a remote extension provider
+ * @alpha
+ */
+export interface RemoteExtensionProviderProps {
+  /** URL where the extension entry point can be loaded from */
+  jsUrl: string;
+  /** URL where the manifest (package.json) can be loaded from */
+  manifestUrl: string;
 }
 
-/** Describes what is needed in order to write an Extension Loader.
+/**
+ * Required props for an extension uploaded to Bentley's Extension Service
  * @alpha
  */
-export interface ExtensionLoader {
-  /** Retrieves an Extension manifest for the provided Extension identifier */
-  getManifest(arg: ExtensionLoaderProps): Promise<ExtensionManifest>;
-  /** Downloads an Extension provided the name of the Extension */
-  downloadExtension(arg: ExtensionLoaderProps): Promise<LocalExtensionProps>;
+export interface ServiceExtensionProviderProps {
+  /** Name of the uploaded extension */
+  name: string;
+  /** Version number (Semantic Versioning) */
+  version: string;
+  /** Context Id */
+  contextId: string;
```

Comment:
This should be switched to iTwinId

Reply:
I can change it in some things but not in the "ExtensionProps" (or what I proposed to call ExtensionMetadata) until we change the actual service api